usl_models/usl_models/flood_ml/model.py

```diff
@@ -118,12 +173,82 @@ def forward(
 
         # Concatenate and feed into remaining ConvLSTM and TransposeConv layers
         # [B, n, H', W', k'] -> [B, H, W, 1]
-        lstm_input = self.concat([st_cnn_output, geo_cnn_output, temp_input])
+        lstm_input = tf.concat([st_cnn_output, geo_cnn_output, temp_input], axis=-1)
         lstm_output = self.conv_lstm(lstm_input)
         output = self.output_cnn(lstm_output)
 
         return output
 
-    def call(self):
-        """Runs the entire autoregressive model."""
-        raise NotImplementedError
+    def call(self, inputs):
+        """Runs the entire autoregressive model.
+
+        Args:
+            inputs: A dictionary of input tensors.
+                - "geospatial" (required): [H, W, f]
+                - "temporal" (required): [T]
+                - "spatiotemporal" (optional): a single flood map [H, W]
+
+        Returns:
+            A tuple (all_flood_predictions, max_flood_map) of the flood predictions and
+            the max predicted flood depth at each cell over the entire storm duration.
+            The maps are the same shape as the input: (H, W).
+        """
+        # TODO: Add data validation
+        geo_input = inputs.get("geospatial")
+        full_temp_input = data_utils.temporal_window_view(
+            inputs.get("temporal"), self.params["m_rainfall"]
+        )
+        # Note: We assume that, if provided, this input is a *single* flood map.
+        # For convenience, we add this to self.flood_maps to align indexing
+        # between flood maps and rainfall ,i.e., the current flooding conditions
+        # and rainfall at time t are stored at index t along the temporal axis.
+        st_input = inputs.get("spatiotemporal")
+        if st_input is None:
+            st_shape = geo_input.shape[:3] + [1]
+            st_input = tf.zeros(st_shape)
+        self.flood_maps.write(0, st_input)
```

Comment:
Is that going to overwrite whatever is at `flood_maps[0]`?
Relatedly, I'm wondering if `flood_maps` should be a local variable within the `call` method rather than an instance variable. What happens if you call `call` twice and inherit a flood map from a previous invocation?

Reply:
There shouldn't be anything at `flood_maps[0]`. My logic is that every index should represent the flood conditions at that time step.

Good point on the multiple `call` invocations. I think this also applies to the geospatial CNN outputs too, and both should be local within a single model call. I'll need a bit more time to double check what should be defined at initialization versus specific runs.

Reply:
Made `flood_maps` a local variable. Also realized that I need to re-set `flood_maps` to the output of the write, since this isn't an in-place operation.

For now, I don't want to worry about whether to pass the `geo_cnn_output` around between various functions and storing it properly, so I'm clearing the output at the end of the model call and setting it back to None. Maybe not the cleanest approach, but I'm still thinking about whether I want to structure the model differently, and I don't want to overcomplicate this PR.